oci/client.py

```diff
@@ -1072,8 +1122,24 @@ def _put_blob_chunked(
             method='POST',
             headers={
                 'content-length': '0',
-            }
+            },
+            raise_for_status=False,
         )
+
+        if (
```

Comment:
:-/

firstly, I think a brief comment explaining this codepath (aws-oci-registry requires repositories to be initialised) would be very helpful for future readers.

secondly, I think we could/should make this slightly more generic, as there are (or were) other oci-registries requiring initialisation (such as AliYun's OCI-Registry).

How about, for example, adding a function (/decorator) that has a semantics of "initialise repository if needed", and have it (for now) contain special-case-code for AWS? I feel this could make the code more expressive, and also easier to extend for other registries.

If taking the decorator-path, maybe this would also help to deduplicate the second codepath below.